src/app/Nutrition/NutritionStateLayer.tsx

```diff
@@ -0,0 +1,200 @@
+import { Button, Dropdown, DropdownItem, DropdownMenu, DropdownTrigger } from '@nextui-org/react';
+import { GeoJsonProperties, Geometry } from 'geojson';
+import { Layer, LeafletMouseEvent, PathOptions } from 'leaflet';
+import React, { useEffect, useRef, useState } from 'react';
+import { GeoJSON } from 'react-leaflet';
+
+import CustomAccordion from '@/components/Accordions/Accordion';
+import LegendContainer from '@/components/Legend/LegendContainer';
+import { GradientLegendContainerItem } from '@/domain/props/GradientLegendContainerItem';
+import NutritionStateChoroplethProps from '@/domain/props/NutritionStateProps';
+
+interface NutrientOption {
+  label: string;
+  key: string;
+}
+
+interface AccordionItem {
+  title: string;
+  iconSrc: string;
+  description: string;
+  content: JSX.Element;
+}
+
+type LayerWithFeature = Layer & {
+  feature?: GeoJSON.Feature<Geometry, GeoJsonProperties>;
+};
+const formatNutrientValue = (value: number | null | undefined): string => {
+  if (value === null || value === undefined) return 'N/A';
+  return `${value.toFixed(1)}%`;
+};
+
+const nutritionFill = (value: number | null): string => {
+  if (!value) return 'none';
+  if (value <= 19) return '#fff3f3';
+  if (value <= 39) return '#fcd0ce';
+  if (value <= 59) return '#f88884';
+  if (value <= 79) return '#f5524c';
+  if (value <= 100) return '#f32e27';
+  return '#A0A0A0';
+};
+
+function StateChoropleth({ regionNutri, regionData, handleClick = () => {}, tooltip }: NutritionStateChoroplethProps) {
+  const [showAccordion] = useState<boolean>(true);
+  const [selectedNutrient, setSelectedNutrient] = useState<string>('mimi_simple');
+  const [selectedLabel, setSelectedLabel] = useState<string>('Select Micronutrient');
+  const layersRef = useRef<LayerWithFeature[]>([]);
+
+  useEffect(() => {
+    layersRef.current.forEach((layer) => {
+      const { feature } = layer;
+      if (feature) {
+        const stateId = feature.id || feature.properties?.id;
+        const match = regionNutri?.features.find((item) => item.id === stateId);
+        const nutrientValue = match ? match?.properties?.nutrition[selectedNutrient] : null;
+        const formattedNutrientValue = formatNutrientValue(nutrientValue);
+        const nutrientLabels: { [key: string]: string } = {
+          fe_ai: 'Iron',
+          fol_ai: 'Folate',
+          va_ai: 'Vitamin A',
+          vb12_ai: 'Vitamin B12',
+          zn_ai: 'Zinc',
+          mimi_simple: 'Mean Adequacy Ratio',
+        };
+        const nutrientLabel = nutrientLabels[selectedNutrient] || 'Unknown Nutrient';
+        const tooltipContent = `
```

Comment:
I think it would be nice to have consistent styling for this tooltip. It should have dark/light bg color (depending on theme), have rounded corners and slightly transparent bg. You can take a look at tooltip styles in my pr